src/versionize_derive/src/fields/union_field.rs

```diff
@@ -121,4 +114,24 @@ impl UnionField {
             _ => panic!("Unsupported field type {:?}", self.ty),
         }
     }
+
+    fn generate_array_deserializer<T: quote::ToTokens>(
```

Comment:
This function's code is  still duplicated, I would either put the fn in common.rs or define a trait ArrayField which would define and implement this function by default and then `impl ArrayField for UnionField {}` and `impl ArrayField for StructField {}`